enhancements/ingress/host-network-bind-options.md

```diff
@@ -0,0 +1,280 @@
+---
+title: ingress-host-network-binding-options
+
+  authors:
+- "@elbehery"
+ 
+reviewers:
+- "@alebedev87"
+- "@frobware"
+- "@Miciah"
+
+approvers:
+- "@knobunc"
+- "@tjungblu"
+
+
+  creation-date: 2022-01-10
+  
+  last-updated: 2022-01-10
+
+  status: implementable
+
+  see-also:
+
+  replaces:
+
+  superseded-by:
+
+---
+
+# Ingress Host Network Binding Options
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement updates the Ingress Controller API to allow specifying binding options in the HostNetwork strategy and allows running many instances of the ingress controller on the same node using the HostNetwork strategy, therefore avoiding port binding conflicts.
+
+## Motivation
+
+While utilising the HostNetwork strategy for ingress controllers, the ports 80, 443, 10443, 10444, 1936 on the host are bound by HAProxy for http, https, no_sni, sni and stats correspondingly. 
+
+Those ports might be occupied by other processes which makes it impossible to run multiple ingress controllers on the same node with HostNetwork strategy. 
+
+### Goals
+
+Enable running multiple set of ingress controllers on the same node with `HostNetwork` endpoint strategy. The approach is to configure the bindings for the following ports:
+- HTTP
+- HTTPS
+- Stats
+
+### Non-Goals
+
+- Allowing configuring custom ports on the ingress controllers that utilise any endpoint publishing strategy other than `HostNetwork`.
+  
+
+- Allowing configuring custom ports on the ingress controllers for `SNI` and `NO_SNI` ports.
+
+
+## Proposal
+
+Allowing configuring binding options on the ingress controllers that use the `HostNetwork` endpoint publishing strategy.
+
+
+Update the IngressController API by adding an optional `BindingOptions` field with type `*IngressControllerBindingOptions` to the `HostNetworkStrategy` struct:
+
+    ```go
+    // HostNetworkStrategy holds parameters for the HostNetwork endpoint publishing
+    // strategy.
+    type HostNetworkStrategy struct {
+        // ...
+        // bindingOptions defines parameters for binding haproxy in ingress controller pods.
+        // All fields are optional and will use their respective defaults if not set.
+        // See specific bindingOptions fields for more details.
+        //
+        //
+        // Setting fields within bindingOptions is generally not recommended. The
+        // default values are suitable for most configurations.
+        //
+        // +optional
+        BindingOptions *IngressControllerBindingOptions `json:"bindingOptions,omitempty"`
+    }
+    ```
```

Comment:
This and subsequent code blocks got rendered weirdly: no syntax highlight and back tips visible.
Maybe it's the matter of indent in the raw markdown, I think the code block doesn't need the indent.

Reply:
done